src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -404,22 +421,24 @@ public KNNVectorFieldType(
             KNNMethodContext knnMethodContext,
             VectorDataType vectorDataType
         ) {
-            this(name, meta, dimension, knnMethodContext, null, vectorDataType);
+            this(name, meta, dimension, knnMethodContext, null, vectorDataType, knnMethodContext.getSpaceType());
         }
 
         public KNNVectorFieldType(
             String name,
             Map<String, String> meta,
             int dimension,
-            KNNMethodContext knnMethodContext,
-            String modelId,
-            VectorDataType vectorDataType
+            @Nullable KNNMethodContext knnMethodContext,
+            @Nullable String modelId,
+            VectorDataType vectorDataType,
+            @Nullable SpaceType spaceType
```

Comment:
why we are making these args as Nullables?

Reply:
these parameters can be null for different mapper, this annotation can make it more clear.